AirLib/include/vehicles/multirotor/firmwares/mavlink/MavLinkMultirotorApi.hpp

```diff
@@ -1209,7 +1209,12 @@ namespace airlib
                     qgc_proxy_ = nullptr;
                 }
                 else {
-                    connection->subscribe([=, this](std::shared_ptr<mavlinkcom::MavLinkConnection> connection_val, const mavlinkcom::MavLinkMessage& msg) {
+                    connection->subscribe([=
+#if __cplusplus >= 202002L
+                                                ,
+                                            this
+#endif
```

Comment:
Why has that become relevant? Seems odd that we only use it for newer versions, when it has been around for older versions before?

Reply:
So the addition of `, this` was only recent: https://github.com/SEESAI/AirSim/commit/888cf52663ad02a0dc447461de1632a8530eed65
And it breaks the Airsim build, as that is C++ 17 (or something older than 20 anyway), where explicit capture of `this` in lambdas raises a warning (and all warnings are errors).
As you probably already knew, taking the `, this` away again breaks SeesInterface2 as in C++ 20, relying on implicit capture raises a warning.
So, this was the solution I found.
William said he wasn't surprised that this hadn't been spotted yet, as he didn't expect that anyone would have tried to re-build AirSim since 5 months ago when you fixed the C++20 build issues.

Reply:
Okay - fair enough. For SI2 builds: we can turn off warnings for individual 3rd party header files, which is what we've done in the past. Airsim is a special case, since it's not maintained anymore, we're the only one updating it anyway, so happy to leave this.